app/javascript/components/login_page/login_form.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux'
+import { FormGroup, ControlLabel, FormControl, HelpBlock, Button, InputGroup, Glyphicon, Alert } from 'react-bootstrap';
+
+import * as actions from '../../store/actions/loginActions';
+import * as validators from '../../store/actions/loginValidationActions';
+
+export class LoginForm extends React.Component {
+
+    handleFormSubmit = (e) => {this.props.dispatch(actions.loginUser(this.props.username, this.props.password, e))};
+
+    render() {
+        let alert;
```

Comment:
Täm ehtohomman vois laittaa omaan funktioonsa, että pysyy render puhtaana niinkuin pitäis. Muistan kyllä itse myös tehneeni näin, en kyllä nyt muista korjasinko jo :D